app/models/generic_event/move_reject.rb

```diff
@@ -15,5 +17,27 @@ def for_feed
         common_feed_attributes['details']['rebook'] = rebook || false
       end
     end
+
+  private
+
+    def notify_move_proposer
+      # If a move is already rejected/cancelled, we don't send the notification
+      # The move status changes *after* this generic_event is created
+      return if move.cancelled?
+
+      email = move_proposed_by_email
+      return unless email
+
+      MoveRejectMailer.notify(email, move, self).deliver_now!
+    end
+
+    def move_proposed_by_email
+      username =
+        move.generic_events
+          .where(type: 'GenericEvent::MoveProposed')
+          .first&.created_by
```

Comment:
created_by can be an email address, I think it would be a good idea to check if this value is an email and if so, just use the email?

Reply:
@Thource Added a check for an email address